typings/index.d.ts

```diff
@@ -138,6 +138,34 @@ declare module 'discord.js' {
     public toJSON(...props: { [key: string]: boolean | string }[]): object;
   }
 
+  interface RouteBuilderMethodOptions {
+    query?: any;
+    versioned?: boolean;
+    auth?: boolean;
+    reason?: string;
+    headers?: {
+      [key: string]: string;
+    };
+    data: any;
+  }
+
+  interface RouteBuilderMethods {
+    get(opts?: RouteBuilderMethodOptions): Promise<any>;
+    post(opts?: RouteBuilderMethodOptions): Promise<any>;
+    delete(opts?: RouteBuilderMethodOptions): Promise<any>;
+    patch(opts?: RouteBuilderMethodOptions): Promise<any>;
+    put(opts?: RouteBuilderMethodOptions): Promise<any>;
```

Comment:
maybe a type parameter for the return type?
```ts
method<T = never>(opts: ...): Promise<T>;```

Reply:
but the request can return any data so 🤷‍♂️

Reply:
thats the point, you can specify it as a type parameter
```ts
const data = await client.api.some.path.get<User>();
```

although i guess
```ts
const data: User = await client.api.some.path.get();
```
would work the same

Reply:
The latter is def a more favorable pattern.

Reply:
TypeScript encourages the latter's pattern - returning `unknown` (here that'd be `Promise<unknown>`), and then leaving the user to cast it.

```typescript
const data = await client.api.some.path.get() as User;
```